src/common/socket.cpp

```diff
@@ -1292,7 +1294,43 @@ int32 recvudp(int32 fd, void* buff, size_t nbytes, int32 flags, struct sockaddr*
 int32 sendudp(int32 fd, void* buff, size_t nbytes, int32 flags, const struct sockaddr* from, socklen_t addrlen)
 {
     TracyZoneScoped;
-    return sSendto(fd, (const char*)buff, (int)nbytes, flags, from, addrlen);
+    int retval = 0;
+
+    // Block while we can't send data to emulate old behavior...
+    while ((retval = sSendto(fd, (const char*)buff, (int)nbytes, flags, from, addrlen)) < 0)
+    {
+        if (retval < 0)
+        {
+            int err = 0;
+#ifdef WIN32_
```

Comment:
I believe you don't need to ifdef this, we have win/unix things like this defined:
```cpp
#define sErrno         WSAGetLastError()
#define S_ENOTSOCK     WSAENOTSOCK
#define S_EWOULDBLOCK  WSAEWOULDBLOCK
#define S_EINTR        WSAEINTR
#define S_ECONNABORTED WSAECONNABORTED

and

#define sErrno         errno
#define S_ENOTSOCK     EBADF
#define S_EWOULDBLOCK  EAGAIN
#define S_EINTR        EINTR
#define S_ECONNABORTED ECONNABORTED
```

So you'd check against `S_ENOTSOCK || S_EWOULDBLOCK`, or somesuch

Reply:
And for OSX in general: if it compiles, assume it's working. It's a secondary platform for us that exists so I can make builds on my Max :D